src/state/analysisGame/analysisGameSaga.ts

```diff
@@ -1,22 +1,55 @@
 import { AnalysisGame, GenericAction } from '../AppState';
 import { put, select, takeEvery } from 'redux-saga/effects';
-import { LOAD_FINISHED_GAME, setAnalysisGame } from './analysisGameActions';
-import { getAnalysisGame } from '../selectors/AnalysisGameStateSelectors';
+import {
+  LOAD_FINISHED_GAME_BY_DATE,
+  LOAD_FINISHED_GAME_BY_ID,
+  LOAD_LATEST_FINISHED_GAME,
+  setAnalysisGame,
+} from './analysisGameActions';
+import {
+  getAnalysisGameByDate,
+  getAnalysisGameById,
+  getLatestAnalysisGame,
+} from '../selectors/analysisGame/AnalysisGameStateSelectors';
 
-function* loadFinishedGame(action: GenericAction) {
-  const finishedGame: AnalysisGame | undefined = yield select(
-    getAnalysisGame,
+function* loadFinishedGameById(action: GenericAction) {
+  const analysisGame: AnalysisGame | undefined = yield select(
+    getAnalysisGameById,
     action.payload,
   );
+
   // TODO if no results in selector: try over network
 
-  if (finishedGame !== undefined) {
-    yield put(setAnalysisGame(finishedGame));
+  if (analysisGame !== undefined) {
+    yield put(setAnalysisGame(analysisGame));
+  }
+}
+
+function* loadFinishedGameByDate(action: GenericAction) {
```

Comment:
These 3 functions do very similar things right now. Should / Can we make this simpler?

Reply:
maybe make the if(){...} in a seperate function. But i think this just makes everything more complicated. I would leave it as is

Reply:
I don't think that would add much value 🤔